src/figuresClasses.ts

```diff
@@ -1,19 +1,88 @@
-export interface Figure {
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  public shape: Shape = 'triangle';
 
-}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (this.a === 0
+      || this.b === 0
+      || this.c === 0
+      || this.a + this.b <= this.c
+      || this.b + this.c <= this.a
+      || this.a + this.c <= this.b
+    ) {
+      throw new Error(
+        `sides ${this.a}, ${this.b} and ${this.c} can't form a triangle`,
+      );
+    }
+  }
 
-export class Circle {
+  getArea(): number {
+    const s: number = 0.5 * (this.a + this.b + this.c);
 
+    return Math.floor(
+      Math.sqrt(s * (s - this.a) * (s - this.b) * (s - this.c)) * 100,
+    ) / 100;
+  }
 }
 
-export class Rectangle {
+export class Circle implements Figure {
+  public shape: Shape = 'circle';
+
+  constructor(
+    public color: Color,
+    public radius: number,
+  ) {
+    if (this.radius <= 0) {
+      throw new Error(
+        `radius ${this.radius} can't form a circle`,
+      );
+    }
+  }
 
+  getArea(): number {
+    return Math.floor(Math.PI * this.radius * this.radius * 100) / 100;
+  }
 }
 
-export function getInfo(figure) {
+export class Rectangle implements Figure {
+  public shape: Shape = 'rectangle';
+
+  constructor(
+    public color: Color,
+    public width: number,
+    public height: number,
+  ) {
+    const diagonal: number = Math.sqrt(this.width ** 2 + this.height ** 2);
+
+    if (diagonal <= this.width
+      || diagonal <= this.height
+      || this.width < 1
+      || this.height < 1
+    ) {
```

Comment:
Let's move to separate method is valid Rectangle

Reply:
fixed